lib/kupong_integration/service.rb

```diff
@@ -0,0 +1,112 @@
+require 'json'
+
+module KupongIntegration  
+  class Service
+    DEFAULT_API_URL = 'https://api.kupong.se/v1.5/coupons'.freeze
+    SEND_PATH = '/send'.freeze
+    
+    PLUS = '+'.freeze
+    CODE = '46'.freeze
+    COUNTRY_CODE = (PLUS + CODE).freeze
+    EMPTY_STRING = ''.freeze
+    
+    DEFAULT_SETTINGS = {}.freeze
+    DEFAULT_API_PATH = ''.freeze
+    
+    SETTINGS_ATTRIBUTES = %i(
+      authorization
+      coupon_id
+    )
+    
+    SUCCESS_CODE = 200
+    CREATED_CODE = 201
+    
+    SANITIZE = '+ -'.freeze
+    
+    attr_reader :settings, :msisdn, :timestamp
+    
+    def self.config(api_url: nil)
+      @@api_url = api_url || DEFAULT_API_URL
+    end
+      
+    def initialize(settings: DEFAULT_SETTINGS, phone:)
+      @settings  = settings
```

Comment:
Don't we need the `authorization` and `coupon_id` to be set? Shouldn't we fail here rather than later on L#97 when we need it?